### PR TITLE
Update and somewhat revise the docs.

### PR DESCRIPTION
**Pull request by ltratt**
The main idea here is to more clearly separate out "building yk" from "configuring yk at run-time".

### CODE REVIEW

File: docs/src/user/install.md [hunk 1]
```diff
@@ -13,13 +13,24 @@ At the time of writing, running Yk requires the following:
 
  - A [Yk-enabled programming language interpreter](interps.md).
 
+ - A recent installation of [Rust](https://www.rust-lang.org/).
```
Comment (vext01) [inline docs/src/user/install.md:16]: Worth adding that it has to be nightly?

File: docs/src/dev/runtime_config.md [hunk 2]
```diff
@@ -78,6 +56,7 @@ then it is implicitly interpreting.
 This variable is only available when building `ykrt` with the
 `yk_jitstate_debug` Cargo feature enabled.
 
+
```
Comment (vext01) [inline docs/src/dev/runtime_config.md:59]: These should have been a separate commit, but I'll let it slide ;)

**Comment by vext01** [comment]
Can we sneak something in about needing a `clangd` in `$PATH` if you want completion for C/C++ code?

ykllvm builds one, but you don't have to use that one.

**Comment by ltratt** [comment]
> Can we sneak something in about needing a clangd in $PATH if you want completion for C/C++ code?

I know nothing about completion, so you'll need to do that in a PR yourself I'm afraid!

File: docs/src/dev/runtime_config.md (hunk 2 above)
Comment (ltratt) [reply docs/src/dev/runtime_config.md:59]: I plead ignorance.

File: docs/src/user/install.md [hunk 3]
```diff
@@ -13,13 +13,24 @@ At the time of writing, running Yk requires the following:
...
 
-Eventually we'd like to distribute source and binary releases of Yk, but we are
-not there yet.
+## Building
+
+Clone the [main yk repository](https://github.com/ykjit/yk) and build
+it with `cargo`:
+
+```sh
+$ git clone --recurse-submodules --depth 1 \
```
[the remark this answers is not in the document]
Comment (ltratt) [reply docs/src/user/install.md:28]: Fixed in 760c844.

**Comment by ltratt** [comment]
https://github.com/ykjit/yk/pull/689/commits/760c844a1bf7312639a286c569bfb0898a6edd7c made me realise that having "install" and "build" is a bit confusing, as many people will assume they're synonyms. I've slightly tweaked the organisation in https://github.com/ykjit/yk/pull/689/commits/b2c83221e743a90fb1d43caa42460f4266d8ab95.

**Comment by vext01** [comment]
> ykllvm builds one, but you don't have to use that one.

I was wrong.

`ykbuild` configures LLVM with `DLLVM_ENABLE_PROJECTS=lld;clang`.

To build `clangd` add `clang-tools-extra`.

**Comment by vext01** [comment]
Please squash

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**